server/src/main/java/org/elasticsearch/health/metadata/HealthMetadata.java

```diff
@@ -126,55 +126,72 @@ public int hashCode() {
         private static final ParseField FLOOD_STAGE_WATERMARK_FIELD = new ParseField("flood_stage_watermark");
+        private static final ParseField FLOOD_STAGE_MAX_HEADROOM_FIELD = new ParseField("flood_stage_max_headroom");
         private static final ParseField FROZEN_FLOOD_STAGE_WATERMARK_FIELD = new ParseField("frozen_flood_stage_watermark");
         private static final ParseField FROZEN_FLOOD_STAGE_MAX_HEADROOM_FIELD = new ParseField("frozen_flood_stage_max_headroom");
-        private static final ParseField HIGH_MAX_HEADROOM_FIELD = new ParseField("high_max_headroom");
-        private static final ParseField FLOOD_STAGE_MAX_HEADROOM_FIELD = new ParseField("flood_stage_max_headroom");
 
         public static final ConstructingObjectParser<Disk, Void> PARSER = new ConstructingObjectParser<>(
             TYPE,
             true,
             (args) -> new Disk(
                 RelativeByteSizeValue.parseRelativeByteSizeValue((String) args[0], HIGH_WATERMARK_FIELD.getPreferredName()),
+                ByteSizeValue.parseBytesSizeValue((String) args[4], HIGH_MAX_HEADROOM_FIELD.getPreferredName()),
                 RelativeByteSizeValue.parseRelativeByteSizeValue((String) args[1], FLOOD_STAGE_WATERMARK_FIELD.getPreferredName()),
+                ByteSizeValue.parseBytesSizeValue((String) args[5], FLOOD_STAGE_MAX_HEADROOM_FIELD.getPreferredName()),
                 RelativeByteSizeValue.parseRelativeByteSizeValue((String) args[2], FROZEN_FLOOD_STAGE_WATERMARK_FIELD.getPreferredName()),
-                ByteSizeValue.parseBytesSizeValue((String) args[3], FROZEN_FLOOD_STAGE_MAX_HEADROOM_FIELD.getPreferredName()),
-                ByteSizeValue.parseBytesSizeValue((String) args[4], HIGH_MAX_HEADROOM_FIELD.getPreferredName()),
-                ByteSizeValue.parseBytesSizeValue((String) args[5], FLOOD_STAGE_MAX_HEADROOM_FIELD.getPreferredName())
+                ByteSizeValue.parseBytesSizeValue((String) args[3], FROZEN_FLOOD_STAGE_MAX_HEADROOM_FIELD.getPreferredName())
             )
         );
 
         static {
             PARSER.declareString(ConstructingObjectParser.constructorArg(), HIGH_WATERMARK_FIELD);
+            PARSER.declareString(ConstructingObjectParser.constructorArg(), HIGH_MAX_HEADROOM_FIELD);
             PARSER.declareString(ConstructingObjectParser.constructorArg(), FLOOD_STAGE_WATERMARK_FIELD);
+            PARSER.declareString(ConstructingObjectParser.constructorArg(), FLOOD_STAGE_MAX_HEADROOM_FIELD);
             PARSER.declareString(ConstructingObjectParser.constructorArg(), FROZEN_FLOOD_STAGE_WATERMARK_FIELD);
             PARSER.declareString(ConstructingObjectParser.constructorArg(), FROZEN_FLOOD_STAGE_MAX_HEADROOM_FIELD);
-            PARSER.declareString(ConstructingObjectParser.constructorArg(), HIGH_MAX_HEADROOM_FIELD);
-            PARSER.declareString(ConstructingObjectParser.constructorArg(), FLOOD_STAGE_MAX_HEADROOM_FIELD);
```

Comment:
So this block of code telling the parser which field will be read in the expected sequence. I think here you need to keep the serialization sequence (I added the index they will have in the argument array when used by the `PARSER` :
```
            PARSER.declareString(ConstructingObjectParser.constructorArg(), HIGH_WATERMARK_FIELD); // #0
            PARSER.declareString(ConstructingObjectParser.constructorArg(), FLOOD_STAGE_WATERMARK_FIELD); // #1
            PARSER.declareString(ConstructingObjectParser.constructorArg(), FROZEN_FLOOD_STAGE_WATERMARK_FIELD); // #2
            PARSER.declareString(ConstructingObjectParser.constructorArg(), FROZEN_FLOOD_STAGE_MAX_HEADROOM_FIELD); // #3
            PARSER.declareString(ConstructingObjectParser.constructorArg(), HIGH_MAX_HEADROOM_FIELD); // #4
            PARSER.declareString(ConstructingObjectParser.constructorArg(), FLOOD_STAGE_MAX_HEADROOM_FIELD); // #5
```

This more like good to know, I will open a PR soon that will remove this code.... There was a simpler way to wire this.